best-practices/client-side/framework/react/components/media-cropper/src/components/media-cropper/index.js

```diff
@@ -0,0 +1,184 @@
+            document.removeEventListener('mouseup', this.mouseUp);
+            document.removeEventListener('mouseleave', this.mouseUp);
+        }
+    }
+
+    click(e) {
+        if (!this.state.moved) {
+            this.refs.fileUploader.click();
+        }
+        this.setState({
+            ...this.state,
+            moved: false
+        })
+    }
+
+    mouseDown(e) {
+        this.setState({
+            ...this.state,
+            dragging: true,
+            mouseStart: {
+                x: e.pageX,
+                y: e.pageY
+            }
+        });
+        e.preventDefault();
+        e.stopPropagation();
+    }
+
+    mouseMove(e) {
+        this.setState({
+            ...this.state,
+            transX: this.state.transX - (this.state.mouseStart.x - e.pageX),
+            transY: this.state.transY - (this.state.mouseStart.y - e.pageY),
+            mouseStart: {
+                x: e.pageX,
+                y: e.pageY
+            },
+            moved: true
+        });
+        e.preventDefault();
+        e.stopPropagation();
+    }
+
+    mouseUp(e) {
+        this.setState({
+            ...this.state,
+            dragging: false
+        });
+        e.preventDefault();
+        e.stopPropagation();
+    }
+
+    mouseWheel(e) {
+        const scale = Math.pow(1.0015, -e.deltaY);
+
+        const mX = e.pageX - e.target.offsetLeft;
+        const mY = e.pageY - e.target.offsetTop;
+
+        let transX = (this.state.transX - mX) * scale + mX;
+        let transY = (this.state.transY - mY) * scale + mY;
+
+        const zoom = this.state.zoom * scale;
+        this.setState({
+            ...this.state,
+            zoom,
+            transX,
+            transY
+        });
+    }
+
+    loadFile(e) {
+        e.preventDefault();
+        e.stopPropagation();
+        const file = e.target.files[0];
+        if (!file) return;
+        if (file.type.startsWith('image/')) {
+            const reader = new FileReader();
+
+            reader.onload = () => {
+                const media = new Image();
+                media.src = reader.result;
+
+                media.onload = () => {
+                    const zoom = Math.max(this.props.width / media.width, this.props.height / media.height);
+                    this.setState({
+                        ...this.state,
+                        media,
+                        transX: 0,
+                        transY: 0,
+                        zoom
+                    });
+                    this.forceUpdate();
+                };
+            };
+
+            reader.readAsDataURL(file);
+        } else {
+            const media = document.createElement('video');
+            media.width = this.props.width;
+            media.height = this.props.height;
+            media.muted = true;
+            media.src = URL.createObjectURL(file);
+            media.currentTime = 10;
+            media.play();
```

Comment:
Will `media.play` automatically start playing the video?

Reply:
yes, it will start loading the video and play it as soon as it is ready; it is used here to force the video to load and render the first the frame (in this case the one at 10 seconds that I set for testing).

Reply:
Can we make this configurable?